drivers/pwm/pwm_gecko.c

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2021 Sun Amar.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT silabs_gecko_pwm
+
+#include <drivers/pwm.h>
+#include <dt-bindings/pwm/pwm.h>
+#include "em_cmu.h"
+#include "em_timer.h"
+
+/** PWM configuration. */
+struct pwm_gecko_config {
+	TIMER_TypeDef *timer;
+	CMU_Clock_TypeDef clock;
+	TIMER_Prescale_TypeDef prescale_enum;
+	uint8_t channel;
+	uint8_t location;
+	uint8_t port;
+	uint8_t pin;
+};
+
+struct pwm_gecko_data {
+	int prescaler;
```

Comment:
Isn't the prescaler a constant? Could it go into `struct pwm_gecko_config` instead?